assets/pages/oneoff-contributions/helpers/ajax.js

```diff
@@ -82,6 +84,30 @@ function requestData(
   });
 }
 
+function postToEndpoint(request: Object, dispatch: Function, abParticipations: Participations): Promise<*> {
+  return fetch(stripeOneOffContributionEndpoint(cookie.get('_test_username')), request).then((response) => {
+    if (response.ok) {
+      successfulConversion(abParticipations);
+      pageView(routes.oneOffContribThankyou, routes.oneOffContribCheckout);
+      dispatch(checkoutSuccess());
+    }
+    return response.json();
+  }).then((responseJson) => {
+    if (responseJson.error.exceptionType === 'CardException') {
```

Comment:
minor: we could pull lines 96-104 into a separate function (handleUnsuccessfulRequest)